src/database/schema/ChannelListSchema.ts

```diff
@@ -514,14 +515,17 @@ class ChannelList implements BaseDbSchema {
     });
   }
 
-  static fromInitAnonymousChatAPI(data: ModifyAnonymousChatData): ChannelList {
+  static fromInitAnonymousChatAPI(
+    data: ModifyAnonymousChatData,
+    type: 'ANON_PM' | 'PM'
+  ): ChannelList {
     return new ChannelList({
       id: data?.message?.cid,
-      channelPicture: '',
+      channelPicture: data?.targetImage || DEFAULT_PROFILE_PIC_PATH,
       name: data?.targetName,
       description: data?.message?.message,
       unreadCount: 0,
-      channelType: 'ANON_PM',
+      channelType: type,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes in this method are significant and improve the flexibility of the function. The addition of the `type` parameter allows the function to handle both 'ANON_PM' and 'PM' types. Also, the use of `DEFAULT_PROFILE_PIC_PATH` as a fallback when `data?.targetImage` is not available is a good practice. However, it's important to ensure that the calling functions are updated to pass the new `type` argument.
```diff
-  static fromInitAnonymousChatAPI(data: ModifyAnonymousChatData): ChannelList {
+  static fromInitAnonymousChatAPI(
+    data: ModifyAnonymousChatData,
+    type: 'ANON_PM' | 'PM'
+  ): ChannelList {
     return new ChannelList({
       id: data?.message?.cid,
-      channelPicture: '',
+      channelPicture: data?.targetImage || DEFAULT_PROFILE_PIC_PATH,
       name: data?.targetName,
       description: data?.message?.message,
       unreadCount: 0,
-      channelType: 'ANON_PM',
+      channelType: type,
       lastUpdatedAt: data?.message?.created_at,
       lastUpdatedBy: data?.message?.user?.id,
       createdAt: data?.message?.created_at,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->